### PR TITLE
Feature/validations

### PR DESCRIPTION
**Pull request by Ayokunnumi1**
These are the project implementations in this branch;

- Used validations for models.
- Wrote unit tests for models.

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+    it 'returns 5 most recent comments for a post' do
+      6.times { |i| Comment.create(post:, user:, text: "Comment #{i + 1}") }
+
+      recent_comments = post.reload.comments.order(created_at: :desc).limit(5)
+
+      expect(recent_comments.count).to eq(5)
+      expect(recent_comments.first.text).to eq('Comment 6')
+      expect(recent_comments.last.text).to eq('Comment 2')
+    end
```
Comment (Strangeal) [inline spec/models/post_spec.rb]: - [ ] Great progress so far! 👏 Your recent work aligns well with the project requirements. However, I noticed that the unit tests for the added validations in the model are currently missing. As per the project guidelines, it's crucial to have comprehensive test coverage to ensure the reliability of your code.

     - Here's a friendly nudge to consider adding tests for the specific validations highlighted in the attached screenshot. 🧐 Each test can be encapsulated within its own `it` block, which not only adheres to best practices but also makes the tests more readable and easy to understand. This small adjustment enhances the overall maintainability of your test suite.

Screenshot 2023-12-09 at 06 39 06

 Happy coding! 🚀👨‍💻😊

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+      # Ensure that the post is created successfully
+      expect(post3).to be_valid
+
+      result = user.recent_posts.to_a
+
+      expect(result).to be_an(Array)
+      expect(result.size).to eq(3)
+      expect(result).to include(post1, post2, post3)
+    end
```
Comment (Strangeal) [inline spec/models/user_spec.rb]: - [ ] As mentioned in my previous comment, I noticed that the unit tests for the added validations in the model are currently missing. As per the project guidelines, it's crucial to have comprehensive test coverage to ensure the reliability of your code.

    - Here's a friendly nudge to consider adding tests for the specific validations highlighted in the attached screenshot.
    
Screenshot 2023-12-09 at 06 49 08

**Review by Strangeal — CHANGES_REQUESTED** [review]
Hi @Ayokunnumi1, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Implement the features ✅ 
- Nice *UI* design ✅ 
- Professional `README` template ✅ 
- Correct `GitHub` workflow ✅ 
- Meaningful **Github** commits ✅ 
- Correct `.github/workflows` for linters ✅ 
- Followed *best-practices* ✅ 
- Well structured project ✅ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  let(:user) { User.create(name: 'Joy', photo: 'https://www.google.com/url', bio: 'she is a software developer', posts_counter: 0) }
+
+  it 'Name should not be blank' do
+    user.name = nil
+    expect(user).to_not be_blank
+  end
```
Comment (gemmen29) [inline spec/models/user_spec.rb:9]: - [ ] Kindly add tests for all validations in the `User` model as requested by the previous reviewer and as per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec.md#:~:text=Add%20unit%20specs%20for%20all%20of%20your%20models%27%20methods%20and%20validations.). Currently there are no tests for `PostsCounter must be an integer greater than or equal to zero.` validation.

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi @Ayokunnumi1, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Highlights💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by frank1738 — APPROVED** [review]
Hi @Ayokunnumi1 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

✅ All models are tested correctly 👍 

## Optional suggestions

N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**